test/should_fail/annotated_types_fail.erl

```diff
@@ -16,7 +16,15 @@ g(X) ->
     N = ?annotate_type(X, integer()),
     atom_or_die(N).
 
-syntax() ->
+-spec horizontal_cast(integer()) -> atom().
+horizontal_cast(N) ->
+    ?assert_type(N, atom()).
+
+syntax_error_1() ->
+    '::'(banana, "not a type").
```

Comment:
I guess one of the syntax errors should call the new `':::'`

Reply:
Yes, that's the line without test coverage.

Reply:
`::` and `:::` are handled by the same code, but there is one test case for `type_check_expr` and another for `type_check_expr_in`.